Jenkinsfile.macos

```diff
@@ -1,175 +1,283 @@
+def utils
+
 pipeline {
 
   agent { label 'macos-v1.4-arm64' }
 
   options {
+    // timeout regardless of activity
+    timeout(activity: false, time: 120)
     timestamps()
     disableConcurrentBuilds()
     buildDiscarder(logRotator(numToKeepStr: '100'))
   }
 
   parameters {
-      string(name: 'RSTUDIO_VERSION_MAJOR',  defaultValue: '1', description: 'RStudio Major Version')
-      string(name: 'RSTUDIO_VERSION_MINOR',  defaultValue: '1', description: 'RStudio Minor Version')
-      string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
-      string(name: 'RSTUDIO_VERSION_SUFFIX', defaultValue: '+9999', description: 'RStudio Pro Suffix Version')
-      string(name: 'GIT_REVISION', defaultValue: 'main', description: 'Git revision to build')
-      string(name: 'BRANCH_NAME', defaultValue: 'origin/main', description: 'Branch name to build')
       string(name: 'SLACK_CHANNEL', defaultValue: '#ide-builds', description: 'Slack channel to publish build message.')
+      string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
+      booleanParam(name: 'DAILY', defaultValue: true, description: 'Runs daily build if true')
```

Comment:
What do you think about defaulting DAILY to false? That way someone has to start a manual daily build from this job with intention (i.e. intentionally skipping the dispatcher)

Reply:
I definitely don't have a strong preference either way--- your logic definitely makes sense, though it sounds like [Craig is leaning more towards `defaultValue: true`](https://github.com/rstudio/rstudio-pro/pull/4241#discussion_r1081477819).

Reply:
Just to close the loop on this comment, I did in fact default `DAILY` to `false` since it seems like that's the way we are leaning.